soc/arc/snps_nsim/Kconfig.soc

```diff
@@ -4,3 +4,4 @@
 config SOC_NSIM
 	bool "Synopsys nSIM simulator for ARC cores"
 	select ARC
+	select CPU_HAS_MPU
```

Comment:
This is not in the right place - we shouldn't select it for all nsim SoCs. Let's move it to `soc/arc/snps_nsim/Kconfig`

```
config SOC_NSIM_HS
	bool "Synopsys ARC HS in nSIM"
	select CPU_HAS_FPU
+	select CPU_HAS_MPU
```